libs/internal/src/serialization/json_flag.cpp

```diff
@@ -208,4 +208,160 @@ tag_invoke(boost::json::value_to_tag<
     return std::make_optional(variation);
 }
 
+namespace data_model {
+void tag_invoke(
+    boost::json::value_from_tag const& unused,
+    boost::json::value& json_value,
+    data_model::Flag::Rollout::WeightedVariation const& weighted_variation) {
+    auto& obj = json_value.emplace_object();
+    obj.emplace("variation", weighted_variation.variation);
+    obj.emplace("weight", weighted_variation.weight);
+
+    WriteMinimal(obj, "untracked", weighted_variation.untracked);
+}
+
+void tag_invoke(boost::json::value_from_tag const& unused,
+                boost::json::value& json_value,
+                data_model::Flag::Rollout::Kind const& kind) {
+    switch (kind) {
+        case Flag::Rollout::Kind::kUnrecognized:
+            // TODO: Should we be preserving the original string.
```

Comment:
I imagine all strongly typed SDKs have this problem. So I will check some others. I am not super happy with it. (Though generally I don't like having to serialize, but it doesn't seem like there is any real alternative.)